### PR TITLE
[MI-3783]: Added the test cases for the plugin configuration on MS Te…

### PR DESCRIPTION
**Pull request by arush-vashishtha**
### Summary
This pr consists the test cases for the following plugin configurations on MS Teams sync plugin,

- Enforce connect account page.
- Sync DM and GM.
- Prompt interval for DM and GM.
- Enabled teams
- Maximum connected users.
- Maximum download size at one time.
- Buffer size for streaming
- Downloading report
- Certificate based subscriptions
- Sync user
- Sync guest user

### CODE REVIEW

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Enforce_connected_accounts.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+---
+
+**Step 1**
+
+1. When the `enforce connected account` option is true and the disconnected user log-in into their account on MM.
+
+**Step 2**
+
+1. When the `enforce connected account` option is true and `Allow to temporarily skip connect user` is true and the disconnected user log-in into their account on MM.
+2. The user click on `Skip for now` option.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Enforce_connected_accounts.md:42]: Rephrase this

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Enforce_connected_accounts.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+**Expected**
+
+The enforce connect page should open and the user should get the option to connect their account to MS Teams.
+After step 2, the user should get directed to their MM account.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Enforce_connected_accounts.md:47]:
```suggestion
After step 2, the user should get directed to their MM account without connecting his account to MS Teams.
```

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Max_connected_user.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+**Step 3**
+
+1. When the `Max Connected Users` option is set to some limit and the count of users (connected atleast once) is equal to the set limit.
+2. Disconnect one user and connect a new user to MS Teams on MM using the `/msteams-sync connect` command.
+
+**Expected**
+
+The new user should be able to connect to their MS Teams account.
+After step 2, the connected users should not get affected.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Max_connected_user.md:53]:
```suggestion
After step 2, the total number of connected users should not get affected.
```

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Max_connected_user.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+1. When the `Max Connected Users` option is set to some limit and the count of users (connected atleast once) is not equal to the set limit.
+2. Connect the new user to MS Teams on MM using the `/msteams-sync connect` command.
+
+**Step 2**
+
+1. When the `Max Connected Users` option is set to some limit and the count of users (connected atleast once) is equal to the set limit.
+2. Edit and decrease the limit of `Max Connected Users` option.
+
+**Step 3**
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Max_connected_user.md:45]: Add the case for connecting any extra user more than the set limit.

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Max_download_size_at_one_time.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. When the `Maximum size of attachments to support complete one time download` size is set and the user click on the attachment to download in the linked channel or DM/GM on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Max_download_size_at_one_time.md:37]: Same, write it as step, not precondition.

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Max_download_size_at_one_time.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+2. Navigate to the downloaded files on your system.
+
+**Expected**
+
+The attachment file should get downloaded.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Max_download_size_at_one_time.md:42]:
```suggestion
The attachment file downloaded should not get affected in any way.
```

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Prompt_interval_DM_GM.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. When the `Prompt interval for DMs and GMs` time is set and disconnected user sends a message in DM/GM from MM just after setting the time or after the interval time is completed.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Prompt_interval_DM_GM.md:37]: Same, fix it.

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Prompt_interval_DM_GM.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+**Expected**
+
+The user should get a message only visible to the disconnected user who send the message regarding to connect their account.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Prompt_interval_DM_GM.md:41]:
```suggestion
The user should get an ephermal post with a message to connect their account to MS Teams.
```

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Sync_DM_and_GM.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. When the `Sync direct and group message` option is enabled send message in the linked DM/GM on MM or MS Teams.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Sync_DM_and_GM.md:37]: Same

Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Sync_DM_and_GM.md:37]:
```suggestion
1. When the `Sync direct and group message` option is enabled send message in the DM/GM on MM or MS Teams.
```

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Sync_DM_and_GM.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+2. Navigate to the linked DM/GM on MS Teams or MM respectively.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Sync_DM_and_GM.md:38]:
```suggestion
2. Navigate to the DM/GM on MS Teams or MM respectively.
```

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Sync_DM_and_GM.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+**Expected**
+
+The uesr should be able to verify the message in the linked DM/GM on MS Teams or MM respectively.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Sync_DM_and_GM.md:42]:
```suggestion
The user should be able to verify the message in the linked DM/GM on MS Teams or MM respectively.
```

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Sync_guest_users.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Set the time in `Sync user` and enable the `Sync guest users` then check the users in system console on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Sync_guest_users.md:37]: Seperate it into 2 different steps.

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Sync_users.md [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+steps_hashed: null
+---
+
+**Step 1**
+
+1. When the time is set in `Sync user` option in the plugin configurations and check users in System console on MM.
+
+**Step 2**
+
+1. After step 1, check for the existing user on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Sync_users.md:41]:
```suggestion
1. After step 1, check for users who already existed on MM.
```

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Enabled_teams.md [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+**Step 2**
+
+1. Add the desired team name in `enabled teams` field in the plugin settings.
+2. Send message in any linked channel of the desired team added on MM or MS Teams.
+2. Navigate to the linked channel in MS Teams or MM respectively.
+
+**Step 3**
+
+1. Add the desired team name in `enabled teams` field in the plugin settings.
+2. Send message in any linked channel of any team other than the added desired team on MM or MS Teams.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Enabled_teams.md:50]:
```suggestion
2. Send message in any linked channel of any team other than the added team on MM or MS Teams.
```

**Review by AayushChaudhary0001 — APPROVED** [review]

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Sync_users.md [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+---
+
+**Step 1**
+
+1. Set the time in `Sync user` option in the plugin settings in MM
+2. Navigate to users in system console on MM.
+
+**Step 2**
+
+1. After step 1, check for users who already existed on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Sync_users.md:42]: do we really need step 2?

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Buffer_size_of_attachment.md
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Buffer_size_of_attachment.md:1]: also please add test cases for the config of auto promotion of synthetic users

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Max_connected_user.md [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+**Step 4**
+
+1. Set the desired limit in `Max Connected Users` option in the plugin settings on MM.
+2. When the count of users (connected atleast once) is equal to the set limit then connect a new user to MS teams on MM using the `/msteams-sync connect` command.
+
+
+**Expected**
+
+The new user should be able to connect to their MS Teams account.
+After step 2, the total number of connected users should not get affected.
```
[the remark this answers is not in the document]
Comment (arush-vashishtha) [reply data/test-cases/plugins/ms-teams-sync/plugin-configurations/Max_connected_user.md:60]: updated the second step as , any of the connected user should not get affected.

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Sync_users.md (hunk 15 above)
Comment (arush-vashishtha) [reply data/test-cases/plugins/ms-teams-sync/plugin-configurations/Sync_users.md:42]: yes to check no existing user is also created as a remote user.

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Certificate_based_subscriptions.md [hunk 17]
```diff
@@ -35,16 +35,16 @@ steps_hashed: null
...
+2. Send a message in the linked channel or DM/GM on MS Teams.
 3. Navigate to the linked channel or DM/GM on MM.
 
 **Expected**
 
-The user should be able to verify the message sync at more pace and their order in the linked channel or DM/GM on MM.
-After step 2, the user should get a delay or there can be a mismatch in the order of the messages synced in the linked channel or DM/GM on MM.
\ No newline at end of file
+The user should be able to verify the message sync at a faster pace in the linked channel or DM/GM on MM.
+After step 2, the user should get a slower pace in the messages sync in the linked channel or DM/GM on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Certificate_based_subscriptions.md:50]:
```suggestion
After step 2, the user should get a message, which can be slower as compared to using certificate based subscriptions in the linked channel or DM/GM on MM.
```

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Buffer_size_of_attachment.md [hunk 18]
```diff
@@ -35,8 +35,8 @@ steps_hashed: null
 **Step 1**
 
 1. Set the `Buffer size for streaming files` in the plugin settings on MM.
-2. Send the media file in the linked channel or DM/GM on MM or MS Teams.
-3. Click on the media file in the linked channel or DM/Gm on MM.
+2. Send the attachment in the linked channel or DM/GM on MM or MS Teams.
+3. Click on the attachment in the linked channel or DM/GM on MM.
 
 **Expected**
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Buffer_size_of_attachment.md:42]: remove word media from result

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Auto_promote_user.md [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Set the `Automatically Promote Synthetic Users` option to `true`, `Synthetic User Auth Service` option to `SAML` or `AD/LDAP` and `Synthetic User Auth Data` option to `ID` or `Mail` or `User Principal Name` in the plugin settings on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Auto_promote_user.md:37]: shouldn't we mention we need to enable sync user job as well?

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Auto_promote_user.md [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+2. Login through SSO on MM platform from a synthetic user.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Auto_promote_user.md:38]: maintain consistency by using `remote` or `synthetic` word

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Auto_promote_user.md [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+**Step 2**
+
+1. Set the `Automatically Promote Synthetic Users` option to `false`, `Synthetic User Auth Service` option to `SAML` or `AD/LDAP` and `Synthetic User Auth Data` option to `ID` or `Mail` or `User Principal Name` in the plugin settings on MM.
+2. Login through SSO on MM platform from a synthetic user.
+
+
+**Expected**
+
+The user should be promoted and get logged-in into their account on MM.
+After step 2, the user shouldn't be promoted nor get logged-in into their account on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Auto_promote_user.md:49]:
```suggestion
After step 2, the user shouldn't be promoted nor get logged-in to their account on MM.
```

File: data/test-cases/plugins/ms-teams-sync/plugin-configurations/Certificate_based_subscriptions.md [hunk 22]
```diff
@@ -47,4 +47,4 @@ steps_hashed: null
 **Expected**
 
 The user should be able to verify the message sync at a faster pace in the linked channel or DM/GM on MM.
-After step 2, the user should get a slower pace in the messages sync in the linked channel or DM/GM on MM.
\ No newline at end of file
+After step 2, the user should get a message, which can be slower as compared to using certificate based subscriptions in the linked channel or DM/GM on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/plugin-configurations/Certificate_based_subscriptions.md:50]:
```suggestion
After step 2, the user should get a message, which can be slower as compared to messages while using certificate based subscriptions in the linked channel or DM/GM on MM.
```

**Review by ayusht2810 — APPROVED** [review]

**Pull request merged**